### ISSUE TITLE
First of all, thank you for your great creation. Is there a 3D data version of MLP-Mixer

### ISSUE THREAD

**Issue by lxy51** [lucidrains/mlp-mixer-pytorch#15]
First of all, thank you for your great creation. Is there a 3D data version of MLP-Mixer

**Comment by lucidrains** [comment]
there is now, in 0.2.0

```python
import torch
from mlp_mixer_pytorch import MLPMixer3D

model = MLPMixer3D(
    image_size = (256, 128),
    time_size = 4,
    time_patch_size = 2,
    channels = 3,
    patch_size = 16,
    dim = 512,
    depth = 12,
    num_classes = 1000
)

video = torch.randn(1, 3, 4, 256, 128)
pred = model(video) # (1, 1000)
```

**Comment by lxy51** [comment]
Hello, where is the 0.2.0 you mentioned? Is it convenient to give a link?

**Comment by lucidrains** [comment]
@lxy51 `pip install mlp-mixer-pytorch -U`

**Comment by lxy51** [comment]
As I am a beginner, I don't quite follow you. Do you mean that you have made MLPMixer3D into a library that we can download and use directly? Where do I look if I want to see the code?

**Comment by lucidrains** [comment]
@lxy51 the code is [here](https://github.com/lucidrains/mlp-mixer-pytorch/blob/main/mlp_mixer_pytorch/mlp_mixer_3d_pytorch.py)

**Comment by lxy51** [comment]
Thank you for your generous help. I see you are a new addition to MLPMixer3D. which flatters me.
Also I have a question. If I enter 3D data that is irregular, say (D, H, W) = (137, 261, 175). Does the code MLPMixer3D still work in this case? Can it dynamically handle irregular data input dimensions?

**Comment by lucidrains** [comment]
@lxy51 mlp mixer only accepts fixed dimension inputs, and you'll need to make sure those dimensions are divisible by the patch size (there are already validation in the code that should give you informative error messages if not)

**Comment by lxy51** [comment]
Ok, thank you. Thank you for your answer.

**Comment by lxy51** [comment]
Sorry to bother you again.
I see that in the def MLPMixer3D you gave, you didn't change (H, W) to (D, H, W).
For example: image_h, image_w = pair(image_size) changed to image_d, image_h, image_w = pair(image_size).
Do these areas need to be changed as well if using the 3D version?

**Comment by lucidrains** [comment]
@lxy51 your `D` will be the `time_size` with the corresponding `time_patch_size`

**Comment by lxy51** [comment]
If I enter 3D data that is irregular, say (D, H, W) = (137, 261, 175). Can I remove the assert judgment statement. Then padding the image (e.g. using zero padding) to make sure that (D, H, W) of the image is divisible by the patch_size. Use def MLPMixer3D by tweaking the code in this way.

**Comment by lucidrains** [comment]
it has to be divisible by the patch size

perhaps you can crop it?

**Comment by lxy51** [comment]
OK, thank you.
